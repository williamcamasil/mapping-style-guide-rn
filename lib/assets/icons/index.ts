export * as Large from './large';
export * as Heavy from './heavy';
export * as Thin from './thin';
export * as Small from './small';
export * as Default from './default';
