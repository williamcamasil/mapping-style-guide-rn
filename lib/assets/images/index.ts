export { default as BtnTakePicture } from './BtnTakePicture';
export { default as BulletPoint } from './BulletPoint';
export { default as CheckboxImg } from './CheckboxImg';
export { default as DialogClose } from './DialogClose/index.svg';
export { default as Lock } from './Lock';
export { default as MaskedCardNumber } from './MaskedCardNumber';
export { default as Pointer } from './Pointer';
export { default as RadioImg } from './RadioImg';
