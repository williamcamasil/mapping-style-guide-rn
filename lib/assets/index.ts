export * as Icons from './icons';
export * as Images from './images';
export * as Lotties from './lotties';
