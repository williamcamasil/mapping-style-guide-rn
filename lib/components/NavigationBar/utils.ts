import { ColorPaletesType } from '../../tokens';

export type NavigationBarVariantType =
  'dark-content' |
  'light-content';

export const getTextColor = (
  variant?: NavigationBarVariantType,
): ColorPaletesType => {
  switch (variant) {
    case 'light-content':
      return 'neutralWhite';
    case 'dark-content':
    default:
      return 'neutralGray700';
  }
};
