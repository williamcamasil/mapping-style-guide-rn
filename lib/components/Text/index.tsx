import React from 'react';
import {
  StyleProp, StyleSheet, Text as RNText, TextProps as RNTextProp, TextStyle,
} from 'react-native';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import {
  ColorPaletesType, FontWeightType, LineHeightType, Typography,
} from '../../tokens';
import {
  getColorStyle, getVariantStyle, getWeightStyle, TextVariantType,
} from './utils';

export type { TextVariantType } from './utils';

export type TextPropTypes = RNTextProp & {
  theme: AppThemeType;
  style?: StyleProp<TextStyle>;
  color?: ColorPaletesType;
  variant?: TextVariantType;
  weight?: FontWeightType;
  lineHeight?: LineHeightType;
};

const styles = StyleSheet.create({
  text: {
    letterSpacing: Typography.letterSpaces.default,
    flexShrink: 1,
  },
});

const Text: React.FC<TextPropTypes> = ({
  children,
  color,
  theme,
  style,
  variant = 'body',
  weight,
  lineHeight,
  allowFontScaling = false,
  ...others
}) => {

  const textStyles = useTextStyles(() => [
    styles.text,
    getColorStyle(theme, color ?? 'neutralGray600'),
    getVariantStyle(theme, variant, lineHeight),
    getWeightStyle(theme, weight),
    style,
  ], [color, lineHeight, style, theme, variant, weight]);

  return (
    <RNText
      {...others}
      allowFontScaling={allowFontScaling}
      style={textStyles}
    >
      {children}
    </RNText>
  );
};

export default withTheme(Text);
