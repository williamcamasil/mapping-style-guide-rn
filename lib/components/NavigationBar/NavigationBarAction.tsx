import React, { ComponentType } from 'react';
import {
  StyleSheet, TouchableOpacity, TouchableOpacityProps,
} from 'react-native';

import { SvgProps } from 'react-native-svg';

import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';
import Spacer from '../Spacer';
import Text from '../Text';
import { NavigationBarVariantType } from './utils';

type NavigationBarActionPropTypes<IconProps extends SvgProps = SvgProps> = TouchableOpacityProps & {
  Icon: ComponentType<IconProps>;
  color?: ColorPaletesType;
  theme: AppThemeType;
  text?: string;
  variant?: NavigationBarVariantType;
};

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
  },
});

const NavigationBarAction: React.FC<NavigationBarActionPropTypes> = ({
  Icon,
  theme,
  color,
  text,
  variant,
  ...others
}) => {
  const getDefaultActionColor = () => {
    if (variant === 'light-content') {
      return 'neutralWhite';
    }
    if (text) {
      return 'neutralGray600';
    }
    return 'neutralGray700';

  };

  const actionColor = color ?? getDefaultActionColor();

  const renderTextRight = () => {
    if (!text) return null;

    return (
      <>
        <Spacer size={theme.spacings.sXXS} />
        <Text
          testID="text-right"
          variant="body"
          color={actionColor}
        >
          {text}
        </Text>
      </>
    );
  };

  return (
    <TouchableOpacity activeOpacity={theme.opacities.intense} style={styles.button} {...others}>
      <Icon width={24} height={24} color={theme.colors[actionColor]} />
      {renderTextRight()}
    </TouchableOpacity>
  );
};

NavigationBarAction.defaultProps = {
  variant: 'dark-content',
};

export default withTheme(NavigationBarAction);
