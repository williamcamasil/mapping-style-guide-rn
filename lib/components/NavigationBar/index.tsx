import React, { useCallback } from 'react';
import {
  Platform,
  SafeAreaView,
  StatusBar, StatusBarProps, StyleSheet, View,
} from 'react-native';

import { useNavigationHolder } from 'mapping-context-rn';

import { Icons } from '../../assets';
import { useDimensions, useTextStyles, useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import Divider from '../Divider';
import Spacer from '../Spacer';
import Text from '../Text';
import NavigationBarAction from './NavigationBarAction';
import { getTextColor, NavigationBarVariantType } from './utils';

const { OS } = Platform;

export type NavigationBarPropsType = {
  children?: React.ReactNode;
  onBackPress?: () => void;
  theme: AppThemeType;
  title?: string;
  rightContent?: React.ReactNode;
  closeIcon?: boolean;
  variant?: NavigationBarVariantType;
  statusBarTranslucent?: StatusBarProps['translucent'];
  statusBarBackgroundColor?: StatusBarProps['backgroundColor'];
  addDivider?: boolean;
};

const styles = StyleSheet.create({
  container: {
    height: 48,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  titleContainer: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    right: 0,
    left: 0,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    textAlign: 'center',
  },
});

type NavigationBarType = React.FC<NavigationBarPropsType> & {
  Action: typeof NavigationBarAction;
};

const NavigationBar: NavigationBarType = ({
  onBackPress, theme, title, rightContent,
  children, closeIcon,
  variant = 'dark-content',
  statusBarTranslucent = true,
  statusBarBackgroundColor = 'transparent',
  addDivider = false,
}) => {
  const navigation = useNavigationHolder();
  const { statusBar } = useDimensions();

  const backButtonTouchableStyles = useTextStyles(() => [
    {
      marginLeft: theme.spacings.sLarge,
    },
  ], [theme]);

  const rightContentStyles = useTextStyles(() => [
    {
      marginRight: theme.spacings.sLarge,
    },
  ], [theme]);

  const childrenContainerStyles = useViewStyles(() => [
    {
      flex: 1,
      paddingHorizontal: theme.spacings.sLarge,
    },
  ], [theme]);

  const safeAreaStyles = useViewStyles(() => [
    {
      paddingTop: OS === 'android' ? statusBar.height : undefined,
    },
  ], [statusBar.height]);

  const handlePress = useCallback(() => {
    if (onBackPress) {
      onBackPress();
      return;
    }

    navigation.goBack();
  }, [navigation, onBackPress]);

  const renderContent = () => {
    if (children) {
      return (
        <View style={childrenContainerStyles}>
          {children}
        </View>
      );
    }

    return (
      <>
        <View style={styles.titleContainer} pointerEvents="none">
          <Text
            style={styles.title}
            variant="body"
            weight="bold"
            lineHeight="large"
            color={getTextColor(variant)}
            numberOfLines={2}
            testID="navigation-bar-title"
          >
            {title}
          </Text>
        </View>

        <View style={rightContentStyles}>
          {rightContent ? (
            rightContent
          ) : null}
        </View>
      </>
    );
  };

  const renderDivider = () => {
    if (!addDivider) return null;

    return (
      <>
        <Spacer size={theme.spacings.sXS} />
        <Divider color="neutralGray100" testID="divider-component" />
      </>
    );
  };

  return (
    <SafeAreaView style={safeAreaStyles}>
      <View style={styles.container}>
        <StatusBar
          barStyle={variant}
          translucent={statusBarTranslucent}
          backgroundColor={statusBarBackgroundColor}
        />
        <NavigationBarAction
          Icon={closeIcon ? Icons.Heavy.Close : Icons.Heavy.ArrowLeft}
          onPress={handlePress}
          style={backButtonTouchableStyles}
          testID="go-back-action"
          variant={variant}
        />
        {renderContent()}
      </View>
      {renderDivider()}
    </SafeAreaView>
  );
};

NavigationBar.Action = NavigationBarAction;

export default withTheme(NavigationBar);
