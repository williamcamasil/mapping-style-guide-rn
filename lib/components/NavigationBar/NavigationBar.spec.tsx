import React from 'react';

import { render, fireEvent } from '@testing-library/react-native';
import { createNavigationMock } from 'mapping-context-rn';
import renderer from 'react-test-renderer';

import NavigationBar from '.';
import { Icons } from '../../assets';
import { DefaultTheme } from '../../theme';
import Text from '../Text';

describe('NavigationBar events', () => {
  it('Should not render the title in component', () => {
    const result = render(<NavigationBar />);

    const titleComponent = result.getByTestId('navigation-bar-title');

    expect(titleComponent.props.children).toBeUndefined();
  });

  it('Should show the title in component', () => {
    const result = render(<NavigationBar title="Teste de título" />);

    const titleComponent = result.getByTestId('navigation-bar-title');

    expect(titleComponent.props.children).toBe('Teste de título');
  });

  it('Should render only the icon back in the left', () => {
    const result = render(<NavigationBar />);

    const buttonBack = result.getByTestId('go-back-action');

    expect(buttonBack).toBeDefined();
  });

  it('Should to render component in right', () => {
    const result = render(<NavigationBar
      title="Teste de título"
      rightContent={(
        <NavigationBar.Action
          testID="question-svg"
          Icon={Icons.Thin.Help}
        />
      )}
    />);

    const component = result.getByTestId('question-svg');

    expect(component).toBeDefined();
  });

  it('Should to render component in right with a text', () => {
    const result = render(<NavigationBar
      title="Teste de título"
      rightContent={(
        <NavigationBar.Action
          testID="question-svg"
          text="Ajuda"
          color="neutralGray600"
          Icon={Icons.Thin.Help}
        />
      )}
    />);

    const component = result.getByTestId('text-right');

    expect(component).toBeDefined();
  });

  it('Should to render component in right with a text color default', () => {
    const result = render(<NavigationBar
      title="Teste de título"
      rightContent={(
        <NavigationBar.Action
          testID="question-svg"
          text="Ajuda"
          Icon={Icons.Thin.Help}
        />
      )}
    />);

    const component = result.getByTestId('text-right');

    expect(component).toBeDefined();
  });

  it('Should to render component with light content', () => {
    const result = render(<NavigationBar
      title="Teste de título"
      variant="light-content"
    />);

    const componentText = result.getByTestId('navigation-bar-title');

    expect(componentText.props.style.color).toBe(DefaultTheme.colors.neutralWhite);
  });

  it('Should call event onBackPress in click of button call the navigation', () => {
    const navigation = createNavigationMock();

    const result = render(<NavigationBar title="Teste de título" />);

    const button = result.getByTestId('go-back-action');

    fireEvent.press(button);

    expect(navigation.goBack).toBeCalled();
  });

  it('Should call event onBackPress in click of button with personal function', () => {
    const backPress = jest.fn();

    const result = render(<NavigationBar title="Teste de título" onBackPress={backPress} />);

    const button = result.getByTestId('go-back-action');

    fireEvent.press(button);

    expect(backPress).toHaveBeenCalled();
  });
});

describe('NavigationBar snapshots', () => {
  it('should match snapshot default', () => {
    const tree = renderer.create(<NavigationBar title="Teste de título" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with close icon', () => {
    const tree = renderer.create(<NavigationBar title="Teste de título" closeIcon />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with children', () => {
    const tree = renderer.create(
      <NavigationBar>
        <Text>
          Qualquer conteúdo
        </Text>
      </NavigationBar>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with light-content', () => {
    const tree = renderer.create(<NavigationBar title="Teste de título" variant="light-content" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom status bar', () => {
    const tree = renderer
      .create(<NavigationBar
        title="Teste de título"
        statusBarTranslucent={false}
        statusBarBackgroundColor="#ff0000"
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot with divider', () => {
    const tree = renderer
      .create(<NavigationBar
        title="Teste de título"
        statusBarTranslucent={false}
        addDivider
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
