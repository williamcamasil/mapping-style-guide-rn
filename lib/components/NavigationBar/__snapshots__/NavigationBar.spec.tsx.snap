// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar snapshots should match snapshot default 1`] = `
<RCTSafeAreaView
  style={
    {
      "paddingTop": undefined,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": 28,
          "opacity": 1,
        }
      }
      testID="go-back-action"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M12.778 3.222a1.1 1.1 0 0 1 0 1.556L5.556 12l7.222 7.222a1.1 1.1 0 1 1-1.555 1.556l-8-8a1.1 1.1 0 0 1 0-1.556l8-8a1.1 1.1 0 0 1 1.555 0Z"
            fill={
              {
                "payload": 4282270025,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          {
            "color": "#3E4149",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
            "textAlign": "center",
          }
        }
        testID="navigation-bar-title"
      >
        Teste de título
      </Text>
    </View>
    <View
      style={
        {
          "marginRight": 28,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`NavigationBar snapshots should match snapshot with children 1`] = `
<RCTSafeAreaView
  style={
    {
      "paddingTop": undefined,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": 28,
          "opacity": 1,
        }
      }
      testID="go-back-action"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M12.778 3.222a1.1 1.1 0 0 1 0 1.556L5.556 12l7.222 7.222a1.1 1.1 0 1 1-1.555 1.556l-8-8a1.1 1.1 0 0 1 0-1.556l8-8a1.1 1.1 0 0 1 1.555 0Z"
            fill={
              {
                "payload": 4282270025,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        {
          "flex": 1,
          "paddingHorizontal": 28,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        Qualquer conteúdo
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`NavigationBar snapshots should match snapshot with close icon 1`] = `
<RCTSafeAreaView
  style={
    {
      "paddingTop": undefined,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": 28,
          "opacity": 1,
        }
      }
      testID="go-back-action"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M4.778 3.222a1.1 1.1 0 0 0-1.556 1.556L10.444 12l-7.222 7.222a1.1 1.1 0 1 0 1.556 1.556L12 13.556l7.222 7.222a1.1 1.1 0 1 0 1.556-1.556L13.556 12l7.222-7.222a1.1 1.1 0 1 0-1.556-1.556L12 10.444 4.778 3.222Z"
            fill={
              {
                "payload": 4282270025,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          {
            "color": "#3E4149",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
            "textAlign": "center",
          }
        }
        testID="navigation-bar-title"
      >
        Teste de título
      </Text>
    </View>
    <View
      style={
        {
          "marginRight": 28,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`NavigationBar snapshots should match snapshot with custom status bar 1`] = `
<RCTSafeAreaView
  style={
    {
      "paddingTop": undefined,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": 28,
          "opacity": 1,
        }
      }
      testID="go-back-action"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M12.778 3.222a1.1 1.1 0 0 1 0 1.556L5.556 12l7.222 7.222a1.1 1.1 0 1 1-1.555 1.556l-8-8a1.1 1.1 0 0 1 0-1.556l8-8a1.1 1.1 0 0 1 1.555 0Z"
            fill={
              {
                "payload": 4282270025,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          {
            "color": "#3E4149",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
            "textAlign": "center",
          }
        }
        testID="navigation-bar-title"
      >
        Teste de título
      </Text>
    </View>
    <View
      style={
        {
          "marginRight": 28,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`NavigationBar snapshots should match snapshot with divider 1`] = `
<RCTSafeAreaView
  style={
    {
      "paddingTop": undefined,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": 28,
          "opacity": 1,
        }
      }
      testID="go-back-action"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M12.778 3.222a1.1 1.1 0 0 1 0 1.556L5.556 12l7.222 7.222a1.1 1.1 0 1 1-1.555 1.556l-8-8a1.1 1.1 0 0 1 0-1.556l8-8a1.1 1.1 0 0 1 1.555 0Z"
            fill={
              {
                "payload": 4282270025,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          {
            "color": "#3E4149",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
            "textAlign": "center",
          }
        }
        testID="navigation-bar-title"
      >
        Teste de título
      </Text>
    </View>
    <View
      style={
        {
          "marginRight": 28,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "height": 16,
        "width": 16,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#F7F7F8",
        "height": 1,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
      }
    }
    testID="divider-component"
  />
</RCTSafeAreaView>
`;

exports[`NavigationBar snapshots should match snapshot with light-content 1`] = `
<RCTSafeAreaView
  style={
    {
      "paddingTop": undefined,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": 28,
          "opacity": 1,
        }
      }
      testID="go-back-action"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M12.778 3.222a1.1 1.1 0 0 1 0 1.556L5.556 12l7.222 7.222a1.1 1.1 0 1 1-1.555 1.556l-8-8a1.1 1.1 0 0 1 0-1.556l8-8a1.1 1.1 0 0 1 1.555 0Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          {
            "color": "#FFFFFF",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
            "textAlign": "center",
          }
        }
        testID="navigation-bar-title"
      >
        Teste de título
      </Text>
    </View>
    <View
      style={
        {
          "marginRight": 28,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
